odl/space/cu_ntuples.py

```diff
@@ -943,8 +1318,14 @@ def norm(self, x):
         norm : float
             The norm of the vector
         """
-        from math import sqrt
-        return sqrt(self.const) * float(_norm_default(x))
+        if self.exponent == 2.0:
+            from math import sqrt
+            return sqrt(self.const) * float(_norm_default(x))
+        elif self.exponent == float('inf'):  # Weighting irrelevant
```

Comment:
No, that's not possible. It's not even the case if `p==2`.

The mathematical behavior we want is that `p==inf` is the limiting case for `p->inf`. In the case of constant weighting, it is easy to see what happens since
```
||x||_{c, p} = sum(c * abs(x)**p)**(1/p) = c**(1/p) ||x||_p
```
so for `p->inf`, the factor `c**(1/p)` goes to 1.

Even in the matrix-weighted case, one can write
```
||x||_{G, p} = ||G**(1/p) x||_p
```
so, by continuity in `p` and since `G**(1/p) -> I` for `p->inf`, the weighted norm goes to the unweighted one in the limiting case. (I've also tested it numerically to be sure :-) )